allennlp/data/fields/index_field.py

```diff
@@ -38,13 +40,15 @@ def get_padding_lengths(self) -> Dict[str, int]:
 
     @overrides
     def as_array(self, padding_lengths: Dict[str, int]) -> numpy.array:
-        one_hot_index = numpy.zeros(padding_lengths['num_options'])
-        one_hot_index[self._index] = 1
-        return one_hot_index
+        index = numpy.zeros(padding_lengths['num_options'])
```

Comment:
Apparently our `IndexField` test doesn't ever call `as_array` - can you add tests for this?